src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -40,33 +40,62 @@
 #include "mediapipegraphdefinition.hpp"  // for version in response
 
 namespace ovms {
-static ov::Tensor bruteForceDeserialize(const std::string& requestedName, const KFSRequest* request) {
+static Status deserializeTensor(const std::string& requestedName, const KFSRequest* request, ov::Tensor& outTensor) {
+    std::string model_version{"1"};  // TODO later
     auto requestInputItr = std::find_if(request->inputs().begin(), request->inputs().end(), [&requestedName](const ::KFSRequest::InferInputTensor& tensor) { return tensor.name() == requestedName; });
     if (requestInputItr == request->inputs().end()) {
-        return ov::Tensor();  // TODO
+        std::stringstream ss;
+        ss << "Required input: " << requestedName;
+        const std::string details = ss.str();
+        SPDLOG_DEBUG("[servable name: {} version: {}] Missing input with specific name - {}", request->model_name(), model_version, details);
+        return Status(StatusCode::INVALID_MISSING_INPUT, details);
     }
-    auto inputIndex = requestInputItr - request->inputs().begin();
-    bool deserializeFromSharedInputContents = request->raw_input_contents().size() > 0;
-    auto bufferLocation = deserializeFromSharedInputContents ? &request->raw_input_contents()[inputIndex] : nullptr;
-    if (!bufferLocation) {
-        SPDLOG_ERROR("Reading buffer for Mediapipe KServe is now only supported when using raw_input_contents");
-        throw 42;
-    }
-    ov::Shape shape;
-    for (int i = 0; i < requestInputItr->shape().size(); i++) {
-        shape.push_back(requestInputItr->shape()[i]);
+    if (request->raw_input_contents().size() == 0 || request->raw_input_contents().size() != request->inputs().size()) {
+        std::stringstream ss;
+        ss << "Cannot find data in raw_input_content for input with name: " << requestedName;
+        const std::string details = ss.str();
+        SPDLOG_DEBUG("[servable name: {} version: {}] Invalid message structure - {}", request->model_name(), model_version, details);
```

Comment:
Maybe implement this now? - handling buffer stored in InferenceInputTensor?

Reply:
Still to be decided if tensor content should be supported as well